frontend/src/components/Library/DynamicTable.tsx

```diff
@@ -7,50 +7,132 @@ import {
-        grid
-        bleed
-        striped
-        dense
-        className="ml-0 mr-0 [--gutter:theme(spacing.6)] "
-      >
-        <TableHead>
-          <TableRow>
-            {data.columns.map((header: string, index: number) => (
-              <TableHeader key={index}>{header}</TableHeader>
-            ))}
-          </TableRow>
-        </TableHead>
-        <TableBody>
-          {dataSubset.map((row: string[] | number[], index: number) => {
-            return (
-              <TableRow key={index}>
-                {row.map((item: string | number, cellIndex: number) => (
-                  <TableCell key={cellIndex} className="font-medium pl-8">
-                    {item}
-                  </TableCell>
+    <div className="relative max-w-7xl border dark:text-gray-300 border-gray-500 bg-gray-800 rounded-xl">
+      <div ref={parent}>
+        {minimized && (
+          <div
+            className="flex items-center justify-between p-2 cursor-pointer"
+            onClick={() => setMinimized(false)}
+          >
+            <div className="ml-2">Data results</div>
+            <CustomTooltip>
+              <button tabIndex={-1}>
+                <ArrowsPointingOutIcon className="w-6 h-6 [&>path]:stroke-[2]" />
+              </button>
+            </CustomTooltip>
+          </div>
+        )}
+
+        {!minimized && (
+          <Table
+            grid
+            bleed
+            striped
+            dense
+            maxRows={limitedView ? 5 : data.rows.length}
+            className="ml-0 mr-0 [--gutter:theme(spacing.6)]"
+          >
+            <TableHead>
+              <TableRow>
+                {data.columns.map((header: string, index: number) => (
+                  <TableHeader key={index}>{header}</TableHeader>
                 ))}
               </TableRow>
-            );
-          })}
-        </TableBody>
-      </Table>
+            </TableHead>
+            <TableBody>
+              {dataSubset.map((row: string[] | number[], index: number) => {
+                return (
+                  <TableRow key={index}>
+                    {row.map((item: string | number, cellIndex: number) => (
+                      <TableCell key={cellIndex} className="font-medium pl-8">
+                        {item}
+                      </TableCell>
+                    ))}
+                  </TableRow>
+                );
+              })}
+            </TableBody>
+          </Table>
+        )}
+      </div>
+
+      <div className="absolute bottom-0 right-0 m-2 flex gap-1">
+        {/* Minimize Icon */}
+
+        {!minimized && (
+          <CustomTooltip hoverText="Minimize">
+            {/* On minimize, also collapse if already expanded */}
+            <button
+              tabIndex={-1}
+              onClick={() => {
+                setMinimized(true);
+                setLimitedView(true);
+              }}
+            >
+              <MinusIcon className="w-6 h-6 [&>path]:stroke-[2]" />
```

Comment:
![image](https://github.com/RamiAwar/dataline/assets/26882839/8212a562-9b64-4db7-93cd-fa98d566591c)
if the table is empty, the icons are a bit outside. We can fix by just not doing the expandable stuff if len rows = 0 or less than eg. 4 rows

Reply:
Even if the table is not empty, there's still an issue when expanding/collapsing
![image](https://github.com/RamiAwar/dataline/assets/26882839/8b6b656c-b0c9-4134-9f3a-2f5082dbb009)
![image](https://github.com/RamiAwar/dataline/assets/26882839/96e2536b-1114-4dc0-8e4a-8175196d6637)
also, I think the icons are flipped i.e. it says "expand" when it's already expanded, collapse when it's already collapsed

Reply:
Another example:
![image](https://github.com/RamiAwar/dataline/assets/26882839/086a15cd-f913-4262-acb3-6e16fe530cd8)
![image](https://github.com/RamiAwar/dataline/assets/26882839/7568f2c1-cf16-432e-aad7-5abb269bd925)

However for tables with more than one row, this issue doesn't seem to be there

Reply:
How are we deciding the "expanded" window height? it's looking to me like we're just adding a bit more height, but I still need to scroll even if it's expanded. eg:
Collapsed: 
![image](https://github.com/RamiAwar/dataline/assets/26882839/64aa4065-69e0-4745-90eb-695b68213c87)

Expanded:
![image](https://github.com/RamiAwar/dataline/assets/26882839/af55f8e7-08ba-4789-8dd8-bcb0fbb845a3)